diff -ur a/thunar/thunar-application.c b/thunar/thunar-application.c
--- a/thunar/thunar-application.c	2019-08-11 03:33:53.000000000 +0300
+++ b/thunar/thunar-application.c	2019-10-16 13:48:55.038790513 +0300
@@ -593,7 +593,7 @@
     /* add missing top border to side pane */
     ".shortcuts-pane { border-top-style: solid; }"
     /* make border thicker during DnD */
-    ".standard-view { border-left-width: 0px; }"
+    ".standard-view { border-left-width: 0px; border-right-width: 0px; }"
     ".standard-view:drop(active) { border-width: 2px; }", -1, NULL);
   screen = gdk_screen_get_default ();
   gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (css_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
