diff -ur a/ebuildtester/docker.py b/ebuildtester/docker.py
--- a/ebuildtester/docker.py	2018-10-08 16:46:28.000000000 +0300
+++ b/ebuildtester/docker.py	2019-11-18 11:57:22.328616053 +0300
@@ -152,10 +152,10 @@
             "--tty",
             "--cap-add", "SYS_ADMIN",
             "--device", "/dev/fuse",
-            "--storage-opt", "size=50G",
             "--workdir", "/root",
-            "--volume", "%s:/usr/portage" % local_portage,
-            "--volume", "/usr/portage/distfiles:/usr/portage/distfiles"]
+            "--volume", "%s:/var/db/repos/gentoo" % local_portage,
+            "--volume", "%s/distfiles:/var/cache/distfiles" % local_portage,
+            "--volume", "%s/packages:/var/cache/binpkgs" % local_portage]
         for o in overlays:
             docker_args += ["--volume=%s:%s" % o]
         docker_args += [docker_image]
@@ -202,6 +202,8 @@
         self.execute(("echo MAKEOPTS=\\\"-j%d\\\" " %
                       (options.options.threads)) +
                      ">> /etc/portage/make.conf")
+       	self.execute("echo EMERGE_DEFAULT_OPTS=\\\"--usepkg --oneshot --keep-going --fail-clean\\\" " +
+                     ">> /etc/portage/make.conf")
         if options.options.unstable:
             self.execute("echo ACCEPT_KEYWORDS=\\\"~amd64\\\" " +
                          ">> /etc/portage/make.conf")
@@ -322,4 +324,4 @@
         self.execute("if [[ -f /etc/portage/package.use/testbuild ]]; then " +
                      "cat /etc/portage/package.use/testbuild; fi")
         self.execute("emerge --info")
-        self.execute("qlop --list")
+        self.execute("qlop")
diff -ur a/ebuildtester/options.py b/ebuildtester/options.py
--- a/ebuildtester/options.py	2018-10-08 16:46:28.000000000 +0300
+++ b/ebuildtester/options.py	2019-11-18 11:51:23.680267273 +0300
@@ -17,7 +17,7 @@
     log_ch.setFormatter(logging.Formatter("%(asctime)s - %(message)s"))
     log.addHandler(log_ch)
 
-    fh = logging.FileHandler("tester.log", "a")
+    fh = logging.FileHandler("/tmp/ebuildtester.log", "w")
     fh.setLevel(logging.INFO)
     fh.setFormatter(logging.Formatter("%(asctime)s - %(message)s"))
     log.addHandler(fh)
diff -ur a/ebuildtester/parse.py b/ebuildtester/parse.py
--- a/ebuildtester/parse.py	2018-10-08 16:46:28.000000000 +0300
+++ b/ebuildtester/parse.py	2019-11-18 11:19:26.140629925 +0300
@@ -87,9 +87,9 @@
     parser.add_argument(
         "--profile",
         help="The profile to use",
-        choices=["default/linux/amd64/17.0",
-                 "default/linux/amd64/17.0/systemd"],
-        default="default/linux/amd64/17.0")
+        choices=["default/linux/amd64/17.1",
+                 "default/linux/amd64/17.1/systemd"],
+        default="default/linux/amd64/17.1")
 
     options = parser.parse_args(args)
 
