--- a/src/xfdesktop-file-icon-manager.c	2017-06-26 16:48:02.000000000 +0300
+++ b/src/xfdesktop-file-icon-manager.c	2018-03-28 17:46:50.308585625 +0300
@@ -1871,17 +1871,6 @@
         
         if(file_icon == fmanager->priv->desktop_icon) {
             /* Menu on the root desktop window */
-            /* show arrange desktop icons option */
-            img = gtk_image_new_from_stock(GTK_STOCK_SORT_ASCENDING, GTK_ICON_SIZE_MENU);
-            gtk_widget_show(img);
-            mi = gtk_image_menu_item_new_with_mnemonic(_("Arrange Desktop _Icons"));
-            gtk_image_menu_item_set_image(GTK_IMAGE_MENU_ITEM(mi), img);
-            gtk_widget_show(mi);
-            gtk_menu_shell_append(GTK_MENU_SHELL(menu), mi);
-            g_signal_connect(G_OBJECT(mi), "activate",
-                             G_CALLBACK(xfdesktop_file_icon_menu_arrange_icons),
-                             fmanager);
-
             /* Desktop settings window */
             img = gtk_image_new_from_stock(GTK_STOCK_PREFERENCES, GTK_ICON_SIZE_MENU);
             gtk_widget_show(img);
